src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -608,7 +608,7 @@ def handle_builtin_gate(
         qubits: List[Union[Identifier, IndexedIdentifier]],
         modifiers: List[QuantumGateModifier],
     ) -> None:
-        """Add unitary operation to the circuit"""
+        """Add unitary operation to the program"""
```

Comment:
I don't think I agree with this change. Here "Program" refers to the object being interpreted and "Circuit" the object being constructed; the operation is being added to the latter

Reply:
I was thinking the same but Kshitij told me it could also support pulse programs in the future. And I did not know much about pulse programs. Hence changed the name to program

https://github.com/aws/amazon-braket-default-simulator-python/pull/162

Should I just change it back to circuit or leave it ?

Reply:
> OpenQASM encapsulates more information than what is described by the circuit class today. So it might be helpful to call this argument Program and pass in a circuit (which is a subset of a program)

The "circuit" here refers to what is built by the interpreter, which with today's architecture, is a level of functionality that we've been calling a circuit. Updates to the simulator to support more of a "program-level" scope are likely to involve a different architecture (ie. not building a single interpreted object). For example the [prototype program simulator](https://github.com/aws/amazon-braket-default-simulator-python/commit/38853b3edac022c9e16c7b7b04b7880ad9ec377a#diff-2e92ff2ca1084c457cd127f2a76a969d0e404853f4f5d74db8a0c8245267ce16R400) doesn't utilize the "circuit-building" part of program context, except for temporarily storing a circuit-level instruction buffer.

Given that context, I think `circuit` makes more sense. Feel free to weight in here @kshitijc